Sources/_CryptoExtras/RSA/RSA.swift

```diff
@@ -0,0 +1,311 @@
+    }
+}
+
+extension _RSA.Signing.PrivateKey {
+    ///  Generates an RSA signature with the given key using the default padding.
+    ///
+    ///  The default padding is PSS using MGF1 with same hash function as produced the digest being
+    ///  signed, and a salt that is as long as the digest. Note that this API will not select any
+    ///  known-insecure digests.
+    ///
+    /// - Parameter digest: The digest to sign.
+    /// - Returns: The RSA Signature.
+    /// - Throws: If there is a failure producing the signature
+    public func signature<D: Digest>(for digest: D) throws -> _RSA.Signing.RSASignature {
+        return try self.signature(for: digest, padding: .PSS)
+    }
+
+    /// Generates an RSA signature with the given key using the default padding.
+    ///
+    /// SHA256 is used as the hash function. The default padding is PSS using MGF1 with SHA256
+    /// and a 32-byte salt.
+    ///
+    /// - Parameter data: The data to sign.
+    /// - Returns: The RSA Signature.
+    /// - Throws: If there is a failure producing the signature.
+    public func signature<D: DataProtocol>(for data: D) throws -> _RSA.Signing.RSASignature {
+        return try self.signature(for: SHA256.hash(data: data), padding: .PSS)
+    }
+
+    ///  Generates an RSA signature with the given key.
+    ///
+    /// - Parameter digest: The digest to sign.
+    /// - Parameter padding: The padding to use.
+    /// - Returns: The RSA Signature.
+    /// - Throws: If there is a failure producing the signature
+    public func signature<D: Digest>(for digest: D, padding: _RSA.Signing.Padding) throws -> _RSA.Signing.RSASignature {
+        return try self.backing.signature(for: digest, padding: padding)
+    }
+
+    /// Generates an RSA signature with the given key.
+    /// SHA256 is used as the hash function.
+    ///
+    /// - Parameter data: The data to sign.
+    /// - Parameter padding: The padding to use.
+    /// - Returns: The RSA Signature.
+    /// - Throws: If there is a failure producing the signature.
+    public func signature<D: DataProtocol>(for data: D, padding: _RSA.Signing.Padding) throws -> _RSA.Signing.RSASignature {
+        return try self.signature(for: SHA256.hash(data: data), padding: padding)
+    }
+ }
+
+extension _RSA.Signing.PublicKey {
+    /// Verifies an RSA signature with the given padding over a given digest using the default padding.
+    ///
+    /// The default padding is PSS using MGF1 with same hash function as produced the digest being
+    /// signed, and a salt that is as long as the digest. Note that this API will not select any
+    /// known-insecure digests.
+    ///
+    /// - Parameters:
+    ///   - signature: The signature to verify
+    ///   - digest: The digest that was signed.
+    /// - Returns: True if the signature is valid, false otherwise.
+    public func isValidSignature<D: Digest>(_ signature: _RSA.Signing.RSASignature, for digest: D) throws -> Bool {
```

Comment:
I'd recommend using different (and probably more than 1 letter) generics names for `D: Digest` and `D: DataProtocol`

Reply:
While I broadly agree, these are in line with the equivalent generic type names for the main Crypto API. I think the consistency is valuable here.

Reply:
I see, makes sense!